src/core/utils/passwordUtils.js

```diff
@@ -0,0 +1,15 @@
+import passwordHash from 'password-hash';
+import jwt from 'jsonwebtoken';
+
+import { INVALID_PASSWORD } from '../../config/messages';
+
+const hashPassword = (password) => passwordHash.generate(password);
+const createToken = (data) => jwt.sign(data, 'shhhhh');
+const verifyPassword = (password, hashedPassword) => new Promise((resolve, reject) => 
```

Comment:
Optional note:
could this function utilise your early created promiseFactory?

Reply:
I was thinking about it, but did not know how I can send error message on reject if I use promiseFactory. Would you suggest me something to do please ?

Reply:
Well, promiseFactory might be changed in a way that the executor function could have the logic you used here where the reject either rejects with actual error or with the custom error.
You're free to try to do it either manually or google for the ready-made solution - it just seems to be a common problem, just hard to account for all the case in the current implementation.

Reply:
I see, actually I was thinking to add extra new argument to pass error. But I wanted to ask your opinion as well. Thanks, this was so helpful!

Reply:
@Dmitry-White , I have used another way. I had issues with using promiseFactory because the verify function itself is boolean. I didnt know how to use reject() in case if return value is false.